mono/metadata/object.c

```diff
@@ -3309,19 +3331,21 @@ mono_copy_value (MonoType *type, void *dest, void *value, int deref_pointer)
  * The value must be in the native format of the field type. 
  */
 void
-mono_field_set_value (MonoObject *obj, MonoClassField *field, void *value)
+mono_field_set_value_internal (MonoObject *obj, MonoClassField *field, void *value)
 {
-	MONO_ENTER_GC_UNSAFE;
```

Comment:
Perhaps this PR should not have any GC mode change changes, i.e. do them in a separate PR.